### PR TITLE
sosreport: add GitHub action to test the role

### LINKED ISSUE #331: sosreport: add GitHub action to test the role with Molecule

### PR DESCRIPTION
**Pull request by ra-rau**
- Add GitHub action to test the role with Molecule
- This closes #331

Signed-off-by: Ramona Rautenberg <rautenberg@osism.tech>

### CODE REVIEW

**Comment by berendt** [comment]
@ra-rau Remove the ``last`` plugin from the list. Looks like this one is not longer available.

**Comment by berendt** [comment]
Not the ``xfs`` plugin. The plugin is really called ``last``.

**Comment by berendt** [comment]
Or try it out in a testbed.

```
docker exec -u root -it osism-ansible /change.sh commons feature/add_test_role_sosreport
```

This way you can checkout this branch inside the osism-ansible container. With ``osism apply sosreport`` you can run the role.

**Comment by berendt** [comment]
Try ``become: false`` for the ``Create local required directories`` task.

```
TASK [osism.commons.sosreport : Create local required directories] *************
Monday 02 May 2022  18:07:57 +0000 (0:00:01.351)       0:00:07.843 ************
fatal: [testbed-node-0.testbed.osism.xyz -> localhost]: FAILED! => {"changed": false, "module_stderr": "/bin/sh: 1: sudo: not found\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 127}
fatal: [testbed-node-1.testbed.osism.xyz -> localhost]: FAILED! => {"changed": false, "module_stderr": "/bin/sh: 1: sudo: not found\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 127}
fatal: [testbed-node-2.testbed.osism.xyz -> localhost]: FAILED! => {"changed": false, "module_stderr": "/bin/sh: 1: sudo: not found\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 127}
fatal: [testbed-manager.testbed.osism.xyz -> localhost]: FAILED! => {"changed": false, "module_stderr": "/bin/sh: 1: sudo: not found\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 127}
```

**Comment by ra-rau** [comment]
Finally, its done. Md5 is deprecated as the default. Now its sha256.

File: roles/sosreport/tasks/main.yml [hunk 1]
```diff
@@ -36,23 +37,29 @@
...
+- name: Fetch sha256 file
   ansible.builtin.fetch:
-    src: "{{ sosreport_tmpdir }}/{{ sosreport_archive_filename }}.md5"
-    dest: "{{ sosreport_archive_directory }}/{{ sosreport_archive_filename }}.md5"
+    src: "{{ sosreport_tmpdir }}/{{ sosreport_archive_filename }}.sha256"
+    dest: "{{ sosreport_archive_directory }}/{{ sosreport_archive_filename }}.sha256"
     flat: true
 
-- name: Fetch archive
+- name: change sos report mode
```
Comment (berendt) [inline roles/sosreport/tasks/main.yml:56]: Use ``Change`` here.

File: roles/sosreport/tasks/main.yml [hunk 2]
```diff
@@ -36,33 +39,47 @@
...
   become: true
+  ansible.builtin.file:
+    path: "{{ item }}"
+    state: directory
+    owner: "{{ operator_user }}"
+    group: "{{ operator_group }}"
+    mode: 0755
+    recurse: true
+  loop:
+    - "{{ sosreport_tmpdir }}"
```
Comment (berendt) [inline roles/sosreport/tasks/main.yml:62]: The loop here makes no sense. I think you can enter this path directly here at Path.

**Review by berendt — CHANGES_REQUESTED** [review]

File: roles/sosreport/tasks/main.yml (hunk 2 above)
Comment (ra-rau) [reply roles/sosreport/tasks/main.yml]: You are right. It makes no sense. I changed it.

**Pull request merged**